setup.py

```diff
@@ -4,53 +4,83 @@
 from __future__ import print_function
 
 from distutils.core import setup, Extension
+import sys
+
 try:
     import commands
 except ImportError:
     import subprocess as commands
-import sys
+run = commands.getstatusoutput
 
 version = '0.12'
 
-def pkgconfig(pkg):
-    def _str2list(pkgstr, onlystr):
+
+class PkgConfigExtension(Extension):
+    '''Extension with lazy properties taken from pkg-config'''
+    def __init__(self, *args, **kwargs):
+        self._pkg = kwargs['pkg']
+        del kwargs['pkg']
+        if 'extra_libraries' in kwargs:
```

Comment:
If it isn't really necessary to delete 'extra_libraries' from kwargs, you can use just `self._extra_libraries = kwargs.get('extra_libraries', [])` instead of condition.